### ISSUE TITLE
DataObject can not handle a DB-view

### ISSUE THREAD

**Issue by p4085f9** [smart-data-lake/smart-data-lake#191]
**Describe the bug**
During the creation of dataObject, if the table.name is actually a name of a DB-view than the assert in JdbcTableDataObject at line 93 fails and throws an exception "($id) Table ${table.fullName} doesn't exist. Define createSQL to create table automatically."

**To Reproduce**
```
dataObjects {
  table_in {
    type = JdbcTableDataObject
    connectionId = xy
    table = {
      name = "table_in"
    }
  }
  table_out {
    type = JdbcTableDataObject
    connectionId = xy
    table = {
      name = "table_out"
    }
  }
}

actions {
  db-view-copy {
    type = CopyAction
    inputId = table_in
    outputId = table_out
    metadata {
     feed = db-view-copy
    }
  }
}
```

**Expected behavior**
It would be good, if SDL could be handle the DB-view too, not just the DB-table.

**Additional context**
Script to create the DB structure:
```
CREATE TABLE TABLE_IN_4_VIEW (A VARCHAR2(1 BYTE));

CREATE TABLE TABLE_OUT (A  VARCHAR2(1 BYTE));

CREATE OR REPLACE VIEW TABLE_IN AS SELECT A FROM TABLE_IN_4_VIEW;

INSERT INTO TABLE_IN_4_VIEW (A) VALUES ('A');
```

**Comment by p4085f9** [comment]
#191: Handle DB-view in DataObject:
- Extended selections with view in method isTableExisting of SQLCatalogs
- Remove not needed uppers

**Comment by pgruetter** [comment]
Also stumbled on this one. I see two ways of fixing it:

1. Extend the check for `OracleSQLCatalog` so `isTableExisting `returns true if it finds a table OR view with the given name. This would also be in line with JDBC whose `INFORMATION_SCHEMA.TABLES` also contains all views. 
1. Create a separate `JdbcViewDataObject` so you can clearly distinguish between table and view in your application.conf. In this case we should extend the check in `DefaultSQLCatalog` so `isTableExisting ` only returns true for tables (not for views). The distinction is possible with the attribute `TABLE_TYPE` which is either `BASE_TABLE` or `VIEW`.

@zzeekk What do you think?

**Comment by zzeekk** [comment]
I would go with 1) as already implemented by Robert.
@p4085f9, can you make a PR with your changes?

**Comment by pgruetter** [comment]
Fixed in #209
